src/sampling.jl

```diff
@@ -527,22 +562,46 @@ function sample!(a::AbstractArray, wv::WeightVec, x::AbstractArray;
 end
 
 sample{T}(a::AbstractArray{T}, wv::WeightVec, n::Integer; replace::Bool=true, ordered::Bool=false) =
-    sample!(a, wv, Array(T, n); replace=replace, ordered=ordered)
+    sample!(a, wv, Array{T}(n); replace=replace, ordered=ordered)
 
 sample{T}(a::AbstractArray{T}, wv::WeightVec, dims::Dims; replace::Bool=true, ordered::Bool=false) =
-    sample!(a, wv, Array(T, dims); replace=replace, ordered=ordered)
+    sample!(a, wv, Array{T}(dims); replace=replace, ordered=ordered)
+
 
 # wsample interface
 
-wsample(w::RealVector) = sample(weights(w))
-wsample(a::AbstractArray, w::RealVector) = sample(a, weights(w))
+"""
+    wsample!(a, w, x, [replace=true,] [ordered=false])
 
+Select a weighted sample from an array `a` and store the result in `x`. The
```

Comment:
"Select a sample from an array `a` with the probabilities for each element given by `w`"? Or something similar.